MonoGame.Framework/Input/IImeService.cs

```diff
@@ -0,0 +1,67 @@
+// MonoGame - Copyright (C) The MonoGame Team
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+using System;
+
+namespace Microsoft.Xna.Framework
+{
+    /// <summary>
+    /// Interface to handle text composition.
+    /// </summary>
+    public interface IImeService
+    {
+        /// <summary>
+        /// Enable the system IMM service to support composited character input.
+        /// This should be called when you expect text input from a user and you support languages
+        /// that require an IME (Input Method Editor).
+        /// </summary>
+        void StartTextInput();
+
+        /// <summary>
+        /// Stop the system IMM service.
+        /// </summary>
+        void StopTextInput();
+
+        /// <summary>
+        /// Show the Ime Candidate window renderer by the OS.
+        /// In SDL2 platform it's always true since there's no API to get candidate texts in SDL2.
+        /// In WindowsDX platform, it's possible to choose show candiate window renderer by the OS or render it yourself.
+        /// </summary>
+        bool ShowOSImeWindow { get; set; }
+
+        /// <summary>
+        /// Position Y of virtual keyboard, for mobile platforms has virtual keyboard.
```

Comment:
```suggestion
        /// Height of the virtual keyboard, for mobile platforms that have a virtual keyboard.
```

Hmm, we have the static [`KeyboardInput`](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Input/KeyboardInput.cs) class that also exposes API to show a virtual keyboard for mobile.
This API seems like it would fit in there better than the IME service.

Reply:
The mobile api in `KeyboardInput` does different things like this ImeService, it creates a popup native Android/iOS view and put a textfield control in it. But ImeService is trying to provide a unified interface to do populate text input events, and sometimes we need to know the virtual keyboard height.